Sources/SwiftCompilerPluginMessageHandling/StandardIOMessageConnection.swift

```diff
@@ -146,6 +152,15 @@ public struct StandardIOMessageConnection: MessageConnection {
   }
 }
 
+#if os(WASI)
+// fatalError writes to stdout, which is the message
+// output stream under WASI
+public func internalError(_ message: String) -> Never {
+  fputs("Internal Error: \(message)\n", _stderr)
```

Comment:
> fatalError writes to stdout

As far as I can see it's `stderr`. Am I missing anything?  https://github.com/apple/swift/blob/77f53a5e50f0cf964ae9dcc42f78ec43227a3db2/stdlib/public/runtime/Errors.cpp#L323-L332

But anyway, I think I'm going to use `fputs("Internal Error: \(message)\n", _stderr)` in all platforms, (we don't want `#file` etc, and other magic in `fatalError`). So I will bring `internalError()` back to `CompilerPlugin.swift`

Reply:
Indeed, looks like you're right. Not sure why I thought it wrote to stdout — might've forgotten to attach stderr to the WASI bridge at some point.